ballerina/tests/test.bal

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/log;
+import ballerina/os;
+import ballerina/test;
+
+configurable boolean isTestOnLiveServer = os:getEnv("IS_TEST_ON_ASGARDEO") == "false";
+
+configurable string serviceUrl = isTestOnLiveServer ? os:getEnv("SERVICE_URL") : "http://localhost:443/t/scim2";
+configurable string tokenUrl = isTestOnLiveServer ? os:getEnv("TOKEN_URL") : "http://localhost:9444/oauth2/token";
+configurable string clientId = isTestOnLiveServer ? os:getEnv("CLIENT_ID") : "dummy";
+configurable string clientSecret = isTestOnLiveServer ? os:getEnv("CLIENT_SECRET") : "dummy";
+
+string createdUserId = "";
+string createdGroupId = "";
+
+Client scim = test:mock(Client);
+
+@test:BeforeSuite
+function initializeClientsForMockServer() returns error? {
+    if isTestOnLiveServer {
+        log:printInfo("Initializing client for Asgardeo server");
+        scim = check new (serviceUrl = serviceUrl,
+            config = {
+                auth: {
+                    tokenUrl: tokenUrl,
+                    clientId: clientId,
+                    clientSecret: clientSecret,
```

Comment:
```suggestion
                    tokenUrl,
                    clientId,
                    clientSecret,
```

Reply:
Fixed in [38ee9db](https://github.com/ballerina-platform/module-ballerinax-scim/pull/20/commits/38ee9db5ea81190fd759d023af7c82a1bd3838c6)